src/utils/actions.js

```diff
@@ -33,13 +33,14 @@ const _paginatedNetworkFetch = async (url, baseUrl, req, parse) => {
     const _fetchNext = async (pageUrl) => {
         const response = await fetch(pageUrl, req);
         if (!response.ok) {
-            throw "Invalid response encountered";
+            const errorData = await parse(response);
```

Comment:
what happens if parsing fails? i.e. not a JSON response - i think this would result in the JSON parse error being the message instead of 'Invalid response encountered' as desired